inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1523,12 +1523,16 @@ void MKLDNNGraphOptimizer::FuseEltwiseAndSimple(MKLDNNGraph &graph) {
     }
 }
 
-void MKLDNNGraphOptimizer::DropDoubleReorders(MKLDNNGraph &graph) {
+void MKLDNNGraphOptimizer::DropRedundantReorders(MKLDNNGraph &graph) {
```

Comment:
I am not sure this is a good idea to extend existing DropDoubleReorders from code readability standpoint. At the same time it allows us to have only one pass through all the graph nodes which better for FIL metric. Maybe it is a good idea to extract different branches into separate functions with meaningful names (like VerticalReorderFusion and HorizontalReorderFusion)

Reply:
> Maybe it is a good idea to extract different branches into separate functions with meaningful names (like VerticalReorderFusion and HorizontalReorderFusion)

Sounds good to me. The function has gotten too long indeed.